src/modules/cdp/peerstatemachine.c

```diff
@@ -1241,8 +1233,12 @@ void Snd_Message(peer *p, AAAMessage *msg)
 	int send_message_before_session_sm = 0;
 	LM_DBG("called to peer [%.*s] for %s with code %d \n", p->fqdn.len,
 			p->fqdn.s, is_req(msg) ? "request" : "response", msg->commandCode);
-	if(msg->sessionId)
+	if(msg->sessionId) {
+		// Ensure proper locking order
+		lock_release(p->lock);
```

Comment:
I am not sure adding lock part here if the right way. I would prefer this function be called with the lock (as mentioned in the comments above at line 1225). I will let other comment on this.